services/slam/client_test.go

```diff
@@ -236,6 +284,13 @@ func TestClientFailingService(t *testing.T) {
 		return nil, errors.New("failure to get internal state")
 	}
 
+	failingSLAMService.GetInternalStateStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+		f := func() ([]byte, error) {
+			return nil, errors.New(" failure during callback")
+		}
+		return f, errors.New("failure during get internal state stream")
```

Comment:
I don't think this is correct. 

if the error is not `nil` then `f` should be `nil`.